src/server/generic_family.cc

```diff
@@ -1437,54 +1440,54 @@ OpResult<void> GenericFamily::OpRen(const OpArgs& op_args, string_view from_key,
                                     bool skip_exists) {
   auto* es = op_args.shard;
   auto& db_slice = es->db_slice();
-  auto [from_it, from_expire] = db_slice.FindExt(op_args.db_cntx, from_key);
-  if (!IsValid(from_it))
+  auto from_res = db_slice.FindMutable(op_args.db_cntx, from_key);
+  if (!IsValid(from_res.it))
     return OpStatus::KEY_NOTFOUND;
 
   if (from_key == to_key)
     return OpStatus::OK;
 
   bool is_prior_list = false;
   DbSlice::AutoUpdater post_updater;
-  auto [to_it, to_expire] = db_slice.FindExt(op_args.db_cntx, to_key);
-  if (IsValid(to_it)) {
+  auto to_res = db_slice.FindMutable(op_args.db_cntx, to_key);
+  if (IsValid(to_res.it)) {
     if (skip_exists)
       return OpStatus::KEY_EXISTS;
 
-    is_prior_list = (to_it->second.ObjType() == OBJ_LIST);
+    is_prior_list = (to_res.it->second.ObjType() == OBJ_LIST);
   }
 
-  bool sticky = from_it->first.IsSticky();
-  uint64_t exp_ts = db_slice.ExpireTime(from_expire);
+  bool sticky = from_res.it->first.IsSticky();
+  uint64_t exp_ts = db_slice.ExpireTime(from_res.exp_it);
 
   // we keep the value we want to move.
-  PrimeValue from_obj = std::move(from_it->second);
+  PrimeValue from_obj = std::move(from_res.it->second);
 
   // Restore the expire flag on 'from' so we could delete it from expire table.
-  from_it->second.SetExpire(IsValid(from_expire));
+  from_res.it->second.SetExpire(IsValid(from_res.exp_it));
 
-  if (IsValid(to_it)) {
-    to_it->second = std::move(from_obj);
-    to_it->second.SetExpire(IsValid(to_expire));  // keep the expire flag on 'to'.
+  if (IsValid(to_res.it)) {
+    to_res.it->second = std::move(from_obj);
+    to_res.it->second.SetExpire(IsValid(to_res.exp_it));  // keep the expire flag on 'to'.
 
     // It is guaranteed that UpdateExpire() call does not erase the element because then
     // from_it would be invalid. Therefore, UpdateExpire does not invalidate any iterators,
     // therefore we can delete 'from_it'.
-    db_slice.UpdateExpire(op_args.db_cntx.db_index, to_it, exp_ts);
-    CHECK(db_slice.Del(op_args.db_cntx.db_index, from_it));
+    db_slice.UpdateExpire(op_args.db_cntx.db_index, to_res.it, exp_ts);
```

Comment:
In OpExpire you call post_updater.Run() berfor you call UpdateExpire. So is it needed to run post udpated before calling UpdateExpire? If so we should add it here as well

Reply:
`UpdateExpire()` can call `Del()` inside, and that's why we need to invoke the post updater before.
But see the comment in the above line: "It is guaranteed that UpdateExpire() call does not erase the element because..."
(if it would delete, then the call to `Del()` below is bad, I guess?)

Reply:
Very confusing...

Reply:
Shall I move the `Run()` call to be before `UpdateExpire()`? I guess there's no harm in that, will it be less confusing?

Reply:
Oh I see here to we call std::move on the value, again here I dont know how this is working

Reply:
What specifically is unclear?